gstreamer/src/query.rs

```diff
@@ -295,20 +382,20 @@ impl<'a> Position<&'a QueryRef> {
         unsafe {
             let mut fmt = mem::uninitialized();
 
-            ffi::gst_query_parse_position(self.as_mut_ptr(), &mut fmt, ptr::null_mut());
+            ffi::gst_query_parse_position(self.0.as_mut_ptr(), &mut fmt, ptr::null_mut());
 
             from_glib(fmt)
         }
     }
 }
 
-impl<'a> Position<&'a mut QueryRef> {
+impl<T: AsMutPtr> Position<T> {
```

Comment:
I think we need to distinguish between things that can be mutable and things that can't. Right now this would allow to call `set()` on a `Position<&QueryRef>` or not? Because the `&mut self` is for `Position`, not `T`.

So this probably needs a `AsMutPtr` and `AsPtr` trait, or something like that

Reply:
Yes, you're right.

This raises a concern though. The getters for the concrete types (e.g. `get_result`) use functions such as `ffi::gst_query_parse_position` which expect *mut ffi::GstQuery` as an argument. So either I cast them all as `*mut ffi::GstQuery` or I make `as_ptr` return a `*mut ffi::GstQuery`, which would seem contradictory in my opinion.

Reply:
The GStreamer C API is not const-correct, like almost no C API :) Always return a mut pointer for both traits

Reply:
Or alternatively just cast the non-mut pointer to a mut one at the place of usage

Reply:
All right!